cli/integration_tests.rs

```diff
@@ -0,0 +1,615 @@
+      let mut p_stdin = process.stdin.take().unwrap();
+      write!(p_stdin, "{}", input).unwrap();
+    }
+
+    // Very important when using pipes: This parent process is still
+    // holding its copies of the write ends, and we have to close them
+    // before we read, otherwise the read end will never report EOF. The
+    // Command object owns the writers now, and dropping it closes them.
+    drop(command);
+
+    let mut actual = String::new();
+    reader.read_to_string(&mut actual).unwrap();
+
+    let status = process.wait().expect("failed to finish process");
+    let exit_code = status.code().unwrap();
+    eprintln!("exit code {:?} {:?}", exit_code, self.exit_code);
+    assert_eq!(self.exit_code, exit_code);
+
+    actual = strip_ansi_codes(&actual).to_string();
+
+    let output_path = root.join(self.output);
+    debug!("output path {}", output_path.display());
+    let expected =
+      std::fs::read_to_string(output_path).expect("cannot read output");
+
+    if !wildcard_match(&expected, &actual) {
+      println!("OUTPUT\n{}\nOUTPUT", actual);
+      println!("EXPECTED\n{}\nEXPECTED", expected);
+      panic!("pattern match failed");
+    }
+  }
+}
+
+fn wildcard_match(pattern: &str, s: &str) -> bool {
+  pattern_match(pattern, s, "[WILDCARD]")
+}
+
+fn pattern_match(pattern: &str, s: &str, wildcard: &str) -> bool {
+  if pattern == wildcard {
+    return true;
+  }
+
+  let parts = pattern.split(wildcard).collect::<Vec<&str>>();
+  if parts.len() == 1 {
+    return pattern == s;
+  }
+
+  if !s.starts_with(parts[0]) {
+    return false;
+  }
+
+  let mut t = s.split_at(parts[0].len());
+
+  for (i, part) in parts.iter().enumerate() {
+    if i == 0 {
+      continue;
+    }
+    dbg!(part, i);
+    if i == parts.len() - 1 && (*part == "" || *part == "\n") {
+      dbg!("exit 1 true", i);
+      return true;
+    }
+    if let Some(found) = t.1.find(*part) {
+      dbg!("found ", found);
+      t = t.1.split_at(found + part.len());
+    } else {
+      dbg!("exit false ", i);
+      return false;
+    }
+  }
+
+  dbg!("end ", t.1.len());
+  t.1.is_empty()
+}
+
+#[test]
+fn test_wildcard_match() {
+  let fixtures = vec![
+    ("foobarbaz", "foobarbaz", true),
+    ("[WILDCARD]", "foobarbaz", true),
+    ("foobar", "foobarbaz", false),
+    ("foo[WILDCARD]baz", "foobarbaz", true),
+    ("foo[WILDCARD]baz", "foobazbar", false),
+    ("foo[WILDCARD]baz[WILDCARD]qux", "foobarbazqatqux", true),
+    ("foo[WILDCARD]", "foobar", true),
+    ("foo[WILDCARD]baz[WILDCARD]", "foobarbazqat", true),
+  ];
+
+  // Iterate through the fixture lists, testing each one
+  for (pattern, string, expected) in fixtures {
+    let actual = wildcard_match(pattern, string);
+    dbg!(pattern, string, expected);
+    assert_eq!(actual, expected);
+  }
+
+  // TODO different wild cards?
+  // assert!(pattern_match("foo[BAR]baz", "foobarbaz", "[BAR]"));
+  // assert!(!pattern_match("foo[BAR]baz", "foobazbar", "[BAR]"));
```

Comment:
Remove these? Seems like you have the test below

Reply:
Yikes, fixed!